vizro-ai/src/vizro_ai/dashboard/_response_models/controls.py

```diff
@@ -0,0 +1,169 @@
+            raise ValueError(
+                """
+                This might be due to the filter target is not found in the controllable components.
+                returning default values.
+                """
+            )
+        return v
+
+    def validate_column(v):
+        """Validate the column."""
+        if v not in df_cols:
+            raise ValueError(f"column must be one of {df_cols}")
+        return v
+
+    @root_validator(allow_reuse=True)
+    def validate_date_picker_column(cls, values):
+        """Validate the column for date picker."""
+        column = values.get("column")
+        selector = values.get("selector")
+        if selector and selector.type == "date_picker":
+            if not pd.api.types.is_datetime64_any_dtype(df_schema[column]):
+                raise ValueError(
+                    f"""
+                    The column '{column}' is not of datetime type. Selector type 'date_picker' is
+                    not allowed. Use 'dropdown' instead.
+                    """
+                )
+        return values
+
+    return create_model(
+        "FilterProxy",
+        targets=(
+            List[str],
+            Field(
+                ...,
+                description=f"""
+                Target component to be affected by filter.
+                Must be one of {controllable_components}. ALWAYS REQUIRED.
+                """,
+            ),
+        ),
+        column=(str, Field(..., description="Column name of DataFrame to filter. ALWAYS REQUIRED.")),
+        __validators__={
+            "validator1": validator("targets", pre=True, each_item=True, allow_reuse=True)(validate_targets),
+            "validator2": validator("column", allow_reuse=True)(validate_column),
+            "validator3": validator("targets", pre=True, allow_reuse=True)(validate_targets_not_empty),
+            "validator4": validate_date_picker_column,
+        },
+        __base__=vm.Filter,
+    )
+
+
+def _create_filter(filter_prompt, model, df_cols, df_schema, controllable_components) -> vm.Filter:
+    result_proxy = _create_filter_proxy(
+        df_cols=df_cols, df_schema=df_schema, controllable_components=controllable_components
+    )
+    proxy = _get_pydantic_output(query=filter_prompt, llm_model=model, response_model=result_proxy, df_info=df_schema)
+    return vm.Filter.parse_obj(
+        proxy.dict(exclude={"selector": {"id": True, "actions": True, "_add_key": True}, "id": True, "type": True})
+    )
+
+
+class ControlPlan(BaseModel):
+    """Control plan model."""
+
+    control_type: CtrlType
+    control_description: str = Field(
+        ...,
+        description="""
+        Description of the control. Include everything that seems to relate to this control.
+        Be as detailed as possible. Keep the original relevant description AS IS. If this control is used
+        to control a specific component, include the relevant component details.
+        """,
+    )
+    df_name: str = Field(
```

Comment:
Does this not have to be the target? Ie the name of the component that this control will target?

Reply:
And validate this against the controllable components?

Reply:
I don't think I fully understand the question here 😆 
Do you mean why we need df_name here not the component id?
we need df_name as the key to access df metadata and retrieve things like column names